codecov.yml

```diff
@@ -7,47 +7,43 @@ coverage:
     # Learn more at https://docs.codecov.io/docs/commit-status
     project:
       default:
-        threshold: 1% # allow this much decrease on project
+        target: auto
+        threshold: 1% # # allow this much decrease on project
+        if_ci_failed: error
+        if_not_found: success
       app:
         target: 70%
-        flags:
-          - app
+        paths:
+          - "app/"
       modules:
         target: 70%
-        flags:
-          - modules
+        paths:
+          - "x/"
+          - "!x/**/client/" # ignore client package
       client:
-        flags:
+        target: 70%
+        paths:
           - client
-    changes: false
+          - "x/**/client/"
 
 comment:
   layout: "reach, diff, files"
   behavior: default # update if exists else create new
   require_changes: true
 
-flags:
-  app:
-    paths:
-      - "app/"
-  modules:
-    paths:
-      - "x/"
-      - "!x/**/client/" # ignore client package
-  client:
-    paths:
-      - "client/"
-      - "x/**/client/"
-
 ignore:
   - "docs"
   - "*.md"
   - "*.rst"
   - "**/*.pb.go"
-  - "types/*.pb.go"
+  - "**/*.pb.gw.go"
   - "tests/*"
   - "tests/**/*"
   - "x/**/*.pb.go"
+  - "x/**/*.pb.gw.go"
+  - "x/**/types/*.pb.go"
+  - "x/**/types/*.pb.gw.go"
```

Comment:
Some lines are redundant:

To ignore all of a file type, no matter where:
`**.*/.pb.go`

To ignore a directory and all its contents:
`tests`